components/raftstore/src/store/fsm/peer.rs

```diff
@@ -630,6 +636,8 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
         self.ctx.has_ready = true;
         if let Some(p) = self.fsm.peer.take_apply_proposals() {
             proposals.push(p);
+        } else {
+            self.fsm.peer.invalid_cache();
```

Comment:
why invalid when there is no proposal. As I understand, it should be invalid when there is a proposal?

Reply:
It is a mistake.